### PR TITLE
Update to Angular9

### PR DESCRIPTION
**Pull request by irustm**
Angular9 just came out.
I think with this update we will reduce the build time for projects that have fullcalendar enabled. 

I think we should release the version @fullcalendar/angular 5.0.0 beta

### CODE REVIEW

**Comment by irustm** [comment]
#270

**Comment by domenichelfenstein** [comment]
When I try to get this to work in my project I get this odd error:
```
Cannot find module '@fullcalendar/angular'.
import { FullCalendarComponent } from "@fullcalendar/angular";
```
But I can confirm the version from angular9-branch has been installed under node_modules.
Any suggestions or has anyone found themselves in the same situation?

**Comment by TejasCMehta** [comment]
@arshaw / @irustm  how can we make this repo upto date and aligned with base full-calendar lib? do we need more contributor or what else can be done?

File: projects/fullcalendar/tsconfig.lib.prod.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.lib.json",
+  "angularCompilerOptions": {
+    "enableIvy": false
```
Comment (Timebutt) [inline projects/fullcalendar/tsconfig.lib.prod.json]: Enabling Ivy is not explicitly required, as it's set to `true` by default ([see official documentation](https://angular.io/guide/angular-compiler-options#enableivy)).

File: angular.json [hunk 2]
```diff
@@ -149,7 +154,12 @@
             "tsConfig": "projects/fullcalendar/tsconfig.lib.json",
             "project": "projects/fullcalendar/ng-package.json"
           }
-        },
+        ,          "configurations": {
```
Comment (Timebutt) [inline angular.json]: Q: I'm guessing this formatting issue is because of copy pasting, or from running the upgrade schematic?

**Comment by Timebutt** [comment]
Is there anything keeping this PR from being merged other than the failing build (which will probably run fine when retried)?

**Comment by domenichelfenstein** [comment]
> Is there anything keeping this PR from being merged other than the failing build (which will probably run fine when retried)?

I've tried to build the branch (locally) and then include my build into my project.
The error I get while AOT-building my angular project stays exactly the same as with the LTS of @fullcalendar/angular

**Comment by Timebutt** [comment]
Ok weird, I'll have a look during the day to see if I can help you out fixing the build ;)

**Comment by bbbenja** [comment]
I think travis issue is resolved by #278

File: projects/fullcalendar/tsconfig.lib.prod.json (hunk 1 above)
Comment (irustm) [reply projects/fullcalendar/tsconfig.lib.prod.json]: By default, this was done by ng-update.

File: angular.json (hunk 2 above)
Comment (irustm) [reply angular.json]: Yeah, it was a schematic.

**Comment by irustm** [comment]
I think everything's compatible.

```
> ng serve


chunk {main} main.js, main.js.map (main) 57.4 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 140 kB [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.15 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 94.6 kB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 3.48 MB [initial] [rendered]
Date: 2020-02-10T20:16:02.640Z - Hash: c44052ccff262946d5e2 - Time: 10522ms
** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
: Compiled successfully.
```

```
ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.1
Node: 13.5.0
OS: darwin x64

Angular: 9.0.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.900.1
@angular-devkit/build-angular      0.900.1
@angular-devkit/build-ng-packagr   0.900.1
@angular-devkit/build-optimizer    0.900.1
@angular-devkit/build-webpack      0.900.1
@angular-devkit/core               9.0.1
@angular-devkit/schematics         9.0.1
@angular/cli                       9.0.1
@ngtools/webpack                   9.0.1
@schematics/angular                9.0.1
@schematics/update                 0.900.1
ng-packagr                         9.0.0
rxjs                               6.5.4
typescript                         3.7.5
webpack                            4.41.2
```

**Comment by irustm** [comment]
But, `ng build fullcalendar-demo --configuration="production"` wrong

```
ERROR in ./projects/fullcalendar/src/lib/fullcalendar.component.ts
Module not found: Error: Can't resolve '@fullcalendar/core/datelib/locale'
```
Field 'browser' doesn't contain a valid alias configuration

**Comment by irustm** [comment]
I think I found something like that.

https://github.com/angular/angular/issues/34027

File: projects/fullcalendar/src/lib/fullcalendar.component.ts [hunk 3]
```diff
@@ -14,25 +14,50 @@ import {
...
+type CustomButtonInput = any;
+type ButtonIconsInput = any;
+type CellInfo = any;
+type ButtonTextCompoundInput = any;
+type ToolbarInput = any;
+type ViewOptionsInput = any;
+type EventSourceInput = any;
+type EventInputTransformer = any;
+type EventSourceErrorResponseHandler = any;
+type EventSourceSuccessResponseHandler = any;
```
Comment (irustm) [inline projects/fullcalendar/src/lib/fullcalendar.component.ts]: @arshaw What do you think, should we make it?

**Comment by domenichelfenstein** [comment]
I've had to change the `package.json` in `projects/fullcalendar`:
```
"dependencies": {
    "@fullcalendar/core": "~4.3.1",
    "@angular/core": "^9.0.0",
    "@angular/common": "^9.0.0",
    "tslib": "^1.10.0",
    "rxjs": "~6.5.4",
    "fast-deep-equal": "^2.0.1"
  }
```

otherwise my project was not able to build...

**Comment by irustm** [comment]
@RicardoFazzi I think @arshaw solve it. He manages it.

**Comment by neryortez** [comment]
Is there anything else to be done or tested...?
I'd like to help if necessary.

**Comment by arshaw** [comment]
the `@fullcalendar/angular@4.4.5-beta` package has been published with all the latest changes from the angular9 branch

**Comment by RicardoFazzi** [comment]
@arshaw it seems to be fine the building but I'm having problems anyway, let me know if anyone can help me with these errors:
```
ERROR in node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:5:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: version, OptionsInput, EventInput, EventDef, EventDefHash, EventInstance, EventInstanceHash, parseEventDef, createEventInstance, EventTuple, BusinessHoursInput, parseBusinessHours, applyAll, debounce, padStart, isInt, capitaliseFirstLetter, parseFieldSpecs, compareByFieldSpecs, compareByFieldSpec, flexibleCompare, computeVisibleDayRange, refineProps, matchCellWidths, uncompensateScroll, compensateScroll, subtractInnerElHeight, isMultiDayRange, distributeHeight, undistributeHeight, preventSelection, allowSelection, preventContextMenu, allowContextMenu, compareNumbers, enableCursor, disableCursor, diffDates, htmlEscape, cssToStr, removeExact, isArraysEqual, memoize, memoizeOutput, memoizeRendering, MemoizedRendering, intersectRects, Rect, pointInsideRect, constrainPoint, getRectCenter, diffPoints, Point, translateRect, mapHash, filterHash, isPropsEqual, findElements, findChildren, htmlToElement, createElement, insertAfterElement, prependToElement, removeElement, appendToElement, applyStyle, applyStyleProp, elementMatches, elementClosest, forceClassName, EventStore, filterEventStoreDefs, createEmptyEventStore, mergeEventStores, getRelevantEvents, eventTupleToStore, EventUiHash, EventUi, processScopedUiProps, combineEventUis, Splitter, SplittableProps, buildGotoAnchorHtml, getAllDayHtml, getDayClasses, preventDefault, listenBySelector, whenTransitionDone, computeInnerRect, computeEdges, computeHeightAndMargins, getClippingParents, computeClippingRect, computeRect, unpromisify, EmitterMixin, EmitterInterface, DateRange, rangeContainsMarker, intersectRanges, rangesEqual, rangesIntersect, rangeContainsRange, Mixin, PositionCache, ScrollComponent, ScrollbarWidths, ScrollController, ElementScrollController, WindowScrollController, Theme, Component, ComponentContext, DateComponent, Seg, EventSegUiInteractionState, Calendar, DatePointTransform, DateSpanTransform, DateSelectionApi, View, ViewProps, FgEventRenderer, buildSegCompareObj, FillRenderer, DateProfileGenerator, DateProfile, ViewDef, ViewSpec, DateSpan, DateSpanApi, DatePointApi, isDateSpansEqual, DateMarker, addDays, startOfDay, addMs, addWeeks, diffWeeks, diffWholeWeeks, diffWholeDays, diffDayAndTime, diffDays, isValidDate, Duration, createDuration, isSingleDay, multiplyDuration, addDurations, asRoughMinutes, asRoughSeconds, asRoughMs, wholeDivideDurations, greatestDurationDenominator, DateEnv, DateMarkerMeta, DateFormatter, createFormatter, VerboseFormattingArg, formatIsoTimeString, NamedTimeZoneImpl, parseMarker, EventSourceDef, EventSource, EventSourceHash, Interaction, InteractionSettings, interactionSettingsToStore, interactionSettingsStore, InteractionSettingsStore, PointerDragEvent, Hit, dateSelectionJoinTransformer, eventDragMutationMassager, EventDropTransformers, EventResizeJoinTransforms, ElementDragging, formatDate, formatRange, globalDefaults, config, RecurringType, ParsedRecurring, DragMetaInput, DragMeta, parseDragMeta, createPlugin, PluginDef, PluginDefInput, ViewPropsTransformer, ViewContainerModifier, reducerFunc, Action, CalendarState, CalendarComponentProps, DayHeader, computeFallbackHeaderFormat, renderDateCell, DaySeries, EventInteractionState, EventRenderRange, sliceEventStore, hasBgRendering, getElSeg, DayTable, DayTableSeg, DayTableCell, Slicer, SlicedProps, EventMutation, applyMutationToEventStore, Constraint, ConstraintInput, AllowFunc, isPropsValid, isInteractionValid, EventApi, requestJson, EventHandlerName, EventHandlerArgs, EventHandlerArg, EventRenderingChoice, NON_DATE_PROPS, DATE_PROPS, eventDefParserFunc, GenericHash, ClassNameInput, ElementContent, UNSCOPED_EVENT_UI_PROPS, default, ThemeClass, EqualityFuncHash, DateComponentHash, CalendarInteraction, CalendarInteractionClass, OptionChangeHandler, OptionChangeHandlerMap, ViewDefHash, ViewSpecHash, DAY_IDS, DurationInput, DateInput, FormatterInput, NamedTimeZoneImplClass, EventSourceError, EventInputTransformer, EventSourceSuccessResponseHandler, EventSourceErrorResponseHandler, EventSourceInput, EventSourceFetcher, InteractionClass, eventIsDraggableTransformer, ElementDraggingClass, rtlDefaults, ViewPropsTransformerClass, PluginSystem, CalendarComponent, eventDefMutationApplier, OverlapFunc, isPropsValidTester, LocaleCodeArg, LocaleSingularArg, RawLocaleMap, OptionsManager, EventSourceApi, ViewClass, ViewConfigInput, ViewConfigInputHash, ViewConfigHash, CmdFormatterFunc, CmdFormatter, FuncFormatterFunc, FuncFormatter, EventSourceFunc, _default, ExternalDefTransform, Toolbar

5 declare module '@fullcalendar/core' {
  ~~~~~~~

  node_modules/@fullcalendar/core/main.d.ts:5:1
    5 declare module '@fullcalendar/core' {
      ~~~~~~~
    Conflicts are in this file.
node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:972:9 - error TS2687: All declarations of 'view' must have identical modifiers.

972         view?: View;
            ~~~~
node_modules/@fullcalendar/core/main.d.ts:5:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: version, OptionsInput, EventInput, EventDef, EventDefHash, EventInstance, EventInstanceHash, parseEventDef, createEventInstance, EventTuple, BusinessHoursInput, parseBusinessHours, applyAll, debounce, padStart, isInt, capitaliseFirstLetter, parseFieldSpecs, compareByFieldSpecs, compareByFieldSpec, flexibleCompare, computeVisibleDayRange, refineProps, matchCellWidths, uncompensateScroll, compensateScroll, subtractInnerElHeight, isMultiDayRange, distributeHeight, undistributeHeight, preventSelection, allowSelection, preventContextMenu, allowContextMenu, compareNumbers, enableCursor, disableCursor, diffDates, htmlEscape, cssToStr, removeExact, isArraysEqual, memoize, memoizeOutput, memoizeRendering, MemoizedRendering, intersectRects, Rect, pointInsideRect, constrainPoint, getRectCenter, diffPoints, Point, translateRect, mapHash, filterHash, isPropsEqual, findElements, findChildren, htmlToElement, createElement, insertAfterElement, prependToElement, removeElement, appendToElement, applyStyle, applyStyleProp, elementMatches, elementClosest, forceClassName, EventStore, filterEventStoreDefs, createEmptyEventStore, mergeEventStores, getRelevantEvents, eventTupleToStore, EventUiHash, EventUi, processScopedUiProps, combineEventUis, Splitter, SplittableProps, buildGotoAnchorHtml, getAllDayHtml, getDayClasses, preventDefault, listenBySelector, whenTransitionDone, computeInnerRect, computeEdges, computeHeightAndMargins, getClippingParents, computeClippingRect, computeRect, unpromisify, EmitterMixin, EmitterInterface, DateRange, rangeContainsMarker, intersectRanges, rangesEqual, rangesIntersect, rangeContainsRange, Mixin, PositionCache, ScrollComponent, ScrollbarWidths, ScrollController, ElementScrollController, WindowScrollController, Theme, Component, ComponentContext, DateComponent, Seg, EventSegUiInteractionState, Calendar, DatePointTransform, DateSpanTransform, DateSelectionApi, View, ViewProps, FgEventRenderer, buildSegCompareObj, FillRenderer, DateProfileGenerator, DateProfile, ViewDef, ViewSpec, DateSpan, DateSpanApi, DatePointApi, isDateSpansEqual, DateMarker, addDays, startOfDay, addMs, addWeeks, diffWeeks, diffWholeWeeks, diffWholeDays, diffDayAndTime, diffDays, isValidDate, Duration, createDuration, isSingleDay, multiplyDuration, addDurations, asRoughMinutes, asRoughSeconds, asRoughMs, wholeDivideDurations, greatestDurationDenominator, DateEnv, DateMarkerMeta, DateFormatter, createFormatter, VerboseFormattingArg, formatIsoTimeString, NamedTimeZoneImpl, parseMarker, EventSourceDef, EventSource, EventSourceHash, Interaction, InteractionSettings, interactionSettingsToStore, interactionSettingsStore, InteractionSettingsStore, PointerDragEvent, Hit, dateSelectionJoinTransformer, eventDragMutationMassager, EventDropTransformers, EventResizeJoinTransforms, ElementDragging, formatDate, formatRange, globalDefaults, config, RecurringType, ParsedRecurring, DragMetaInput, DragMeta, parseDragMeta, createPlugin, PluginDef, PluginDefInput, ViewPropsTransformer, ViewContainerModifier, reducerFunc, Action, CalendarState, CalendarComponentProps, DayHeader, computeFallbackHeaderFormat, renderDateCell, DaySeries, EventInteractionState, EventRenderRange, sliceEventStore, hasBgRendering, getElSeg, DayTable, DayTableSeg, DayTableCell, Slicer, SlicedProps, EventMutation, applyMutationToEventStore, Constraint, ConstraintInput, AllowFunc, isPropsValid, isInteractionValid, EventApi, requestJson, EventHandlerName, EventHandlerArgs, EventHandlerArg, EventRenderingChoice, NON_DATE_PROPS, DATE_PROPS, eventDefParserFunc, GenericHash, ClassNameInput, ElementContent, UNSCOPED_EVENT_UI_PROPS, default, ThemeClass, EqualityFuncHash, DateComponentHash, CalendarInteraction, CalendarInteractionClass, OptionChangeHandler, OptionChangeHandlerMap, ViewDefHash, ViewSpecHash, DAY_IDS, DurationInput, DateInput, FormatterInput, NamedTimeZoneImplClass, EventSourceError, EventInputTransformer, EventSourceSuccessResponseHandler, EventSourceErrorResponseHandler, EventSourceInput, EventSourceFetcher, InteractionClass, eventIsDraggableTransformer, ElementDraggingClass, rtlDefaults, ViewPropsTransformerClass, PluginSystem, CalendarComponent, eventDefMutationApplier, OverlapFunc, isPropsValidTester, LocaleCodeArg, LocaleSingularArg, RawLocaleMap, OptionsManager, EventSourceApi, ViewClass, ViewConfigInput, ViewConfigInputHash, ViewConfigHash, CmdFormatterFunc, CmdFormatter, FuncFormatterFunc, FuncFormatter, EventSourceFunc, _default, ExternalDefTransform, Toolbar

5 declare module '@fullcalendar/core' {
 [39m[0m ~~~~~~~

  node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:5:1
    5 declare module '@fullcalendar/core' {
      ~~~~~~~
    Conflicts are in this file.
node_modules/@fullcalendar/core/main.d.ts:115:9 - error TS2374: Duplicate string index signature.

115         [viewId: string]: string | undefined;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:453:9 - error TS2374: Duplicate string index signature.

453         [extendedProp: string]: any;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:970:9 - error TS2687: All declarations of 'view' must have identical modifiers.

970         view: View;
            ~~~~
node_modules/@fullcalendar/core/main.d.ts:1009:9 - error TS2717: Subsequent property declarations must have the same type.  Property 'component' must be of type 'DateComponent<any>', but here has type 'DateComponent<any>'.

1009         component?: DateComponent<any>;
             ~~~~~~~~~

  node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:1012:9
    1012         component?: DateComponent<any>;
                 ~~~~~~~~~
    'component' was also declared here.
node_modules/@fullcalendar/core/main.d.ts:1014:9 - error TS2374: Duplicate string index signature.

1014         [otherProp: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:1494:9 - error TS2374: Duplicate string index signature.

1494         [otherProp: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:1503:9 - error TS2374: Duplicate string index signature.

1503         [otherProp: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:1799:9 - error TS2374: Duplicate string index signature.

1799         [otherProp: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:2580:9 - error TS2374: Duplicate string index signature.

2580         [otherProp: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@fullcalendar/core/main.d.ts:2631:9 - error TS2374: Duplicate string index signature.

2631         [optionName: string]: any;
             ~~~~~~~~~~~~~~~~~~~~~~~~~~

```



```
  ERROR in node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:5:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: version, OptionsInput, EventInput, EventDef, EventDefHash, EventInstance, EventInstanceHash, parseEventDef, createEventInstance, EventTuple, BusinessHoursInput, parseBusinessHours, applyAll, debounce, padStart, isInt, capitaliseFirstLetter, parseFieldSpecs, compareByFieldSpecs, compareByFieldSpec, flexibleCompare, computeVisibleDayRange, refineProps, matchCellWidths, uncompensateScroll, compensateScroll, subtractInnerElHeight, isMultiDayRange, distributeHeight, undistributeHeight, preventSelection, allowSelection, preventContextMenu, allowContextMenu, compareNumbers, enableCursor, disableCursor, diffDates, htmlEscape, cssToStr, removeExact, isArraysEqual, memoize, memoizeOutput, memoizeRendering, MemoizedRendering, intersectRects, Rect, pointInsideRect, constrainPoint, getRectCenter, diffPoints, Point, translateRect, mapHash, filterHash, isPropsEqual, findElements, findChildren, htmlToElement, createElement, insertAfterElement, prependToElement, removeElement, appendToElement, applyStyle, applyStyleProp, elementMatches, elementClosest, forceClassName, EventStore, filterEventStoreDefs, createEmptyEventStore, mergeEventStores, getRelevantEvents, eventTupleToStore, EventUiHash, EventUi, processScopedUiProps, combineEventUis, Splitter, SplittableProps, buildGotoAnchorHtml, getAllDayHtml, getDayClasses, preventDefault, listenBySelector, whenTransitionDone, computeInnerRect, computeEdges, computeHeightAndMargins, getClippingParents, computeClippingRect, computeRect, unpromisify, EmitterMixin, EmitterInterface, DateRange, rangeContainsMarker, intersectRanges, rangesEqual, rangesIntersect, rangeContainsRange, Mixin, PositionCache, ScrollComponent, ScrollbarWidths, ScrollController, ElementScrollController, WindowScrollController, Theme, Component, ComponentContext, DateComponent, Seg, EventSegUiInteractionState, Calendar, DatePointTransform, DateSpanTransform, DateSelectionApi, View, ViewProps, FgEventRenderer, buildSegCompareObj, FillRenderer, DateProfileGenerator, DateProfile, ViewDef, ViewSpec, DateSpan, DateSpanApi, DatePointApi, isDateSpansEqual, DateMarker, addDays, startOfDay, addMs, addWeeks, diffWeeks, diffWholeWeeks, diffWholeDays, diffDayAndTime, diffDays, isValidDate, Duration, createDuration, isSingleDay, multiplyDuration, addDurations, asRoughMinutes, asRoughSeconds, asRoughMs, wholeDivideDurations, greatestDurationDenominator, DateEnv, DateMarkerMeta, DateFormatter, createFormatter, VerboseFormattingArg, formatIsoTimeString, NamedTimeZoneImpl, parseMarker, EventSourceDef, EventSource, EventSourceHash, Interaction, InteractionSettings, interactionSettingsToStore, interactionSettingsStore, InteractionSettingsStore, PointerDragEvent, Hit, dateSelectionJoinTransformer, eventDragMutationMassager, EventDropTransformers, EventResizeJoinTransforms, ElementDragging, formatDate, formatRange, globalDefaults, config, RecurringType, ParsedRecurring, DragMetaInput, DragMeta, parseDragMeta, createPlugin, PluginDef, PluginDefInput, ViewPropsTransformer, ViewContainerModifier, reducerFunc, Action, CalendarState, CalendarComponentProps, DayHeader, computeFallbackHeaderFormat, renderDateCell, DaySeries, EventInteractionState, EventRenderRange, sliceEventStore, hasBgRendering, getElSeg, DayTable, DayTableSeg, DayTableCell, Slicer, SlicedProps, EventMutation, applyMutationToEventStore, Constraint, ConstraintInput, AllowFunc, isPropsValid, isInteractionValid, EventApi, requestJson, EventHandlerName, EventHandlerArgs, EventHandlerArg, EventRenderingChoice, NON_DATE_PROPS, DATE_PROPS, eventDefParserFunc, GenericHash, ClassNameInput, ElementContent, UNSCOPED_EVENT_UI_PROPS, default, ThemeClass, EqualityFuncHash, DateComponentHash, CalendarInteraction, CalendarInteractionClass, OptionChangeHandler, OptionChangeHandlerMap, ViewDefHash, ViewSpecHash, DAY_IDS, DurationInput, DateInput, FormatterInput, NamedTimeZoneImplClass, EventSourceError, EventInputTransformer, EventSourceSuccessResponseHandler, EventSourceErrorResponseHandler, EventSourceInput, EventSourceFetcher, InteractionClass, eventIsDraggableTransformer, ElementDraggingClass, rtlDefaults, ViewPropsTransformerClass, PluginSystem, CalendarComponent, eventDefMutationApplier, OverlapFunc, isPropsValidTester, LocaleCodeArg, LocaleSingularArg, RawLocaleMap, OptionsManager, EventSourceApi, ViewClass, ViewConfigInput, ViewConfigInputHash, ViewConfigHash, CmdFormatterFunc, CmdFormatter, FuncFormatterFunc, FuncFormatter, EventSourceFunc, _default, ExternalDefTransform, Toolbar
    
    5 declare module '@fullcalendar/core' {
      ~~~~~~~
    
      node_modules/@fullcalendar/core/main.d.ts:5:1
        5 declare module '@fullcalendar/core' {
          ~~~~~~~
        Conflicts are in this file.
    node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:972:9 - error TS2687: All declarations of 'view' must have identical modifiers.
    
    972         view?: View;
                ~~~~
    node_modules/@fullcalendar/core/main.d.ts:5:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: version, OptionsInput, EventInput, EventDef, EventDefHash, EventInstance, EventInstanceHash, parseEventDef, createEventInstance, EventTuple, BusinessHoursInput, parseBusinessHours, applyAll, debounce, padStart, isInt, capitaliseFirstLetter, parseFieldSpecs, compareByFieldSpecs, compareByFieldSpec, flexibleCompare, computeVisibleDayRange, refineProps, matchCellWidths, uncompensateScroll, compensateScroll, subtractInnerElHeight, isMultiDayRange, distributeHeight, undistributeHeight, preventSelection, allowSelection, preventContextMenu, allowContextMenu, compareNumbers, enableCursor, disableCursor, diffDates, htmlEscape, cssToStr, removeExact, isArraysEqual, memoize, memoizeOutput, memoizeRendering, MemoizedRendering, intersectRects, Rect, pointInsideRect, constrainPoint, getRectCenter, diffPoints, Point, translateRect, mapHash, filterHash, isPropsEqual, findElements, findChildren, htmlToElement, createElement, insertAfterElement, prependToElement, removeElement, appendToElement, applyStyle, applyStyleProp, elementMatches, elementClosest, forceClassName, EventStore, filterEventStoreDefs, createEmptyEventStore, mergeEventStores, getRelevantEvents, eventTupleToStore, EventUiHash, EventUi, processScopedUiProps, combineEventUis, Splitter, SplittableProps, buildGotoAnchorHtml, getAllDayHtml, getDayClasses, preventDefault, listenBySelector, whenTransitionDone, computeInnerRect, computeEdges, computeHeightAndMargins, getClippingParents, computeClippingRect, computeRect, unpromisify, EmitterMixin, EmitterInterface, DateRange, rangeContainsMarker, intersectRanges, rangesEqual, rangesIntersect, rangeContainsRange, Mixin, PositionCache, ScrollComponent, ScrollbarWidths, ScrollController, ElementScrollController, WindowScrollController, Theme, Component, ComponentContext, DateComponent, Seg, EventSegUiInteractionState, Calendar, DatePointTransform, DateSpanTransform, DateSelectionApi, View, ViewProps, FgEventRenderer, buildSegCompareObj, FillRenderer, DateProfileGenerator, DateProfile, ViewDef, ViewSpec, DateSpan, DateSpanApi, DatePointApi, isDateSpansEqual, DateMarker, addDays, startOfDay, addMs, addWeeks, diffWeeks, diffWholeWeeks, diffWholeDays, diffDayAndTime, diffDays, isValidDate, Duration, createDuration, isSingleDay, multiplyDuration, addDurations, asRoughMinutes, asRoughSeconds, asRoughMs, wholeDivideDurations, greatestDurationDenominator, DateEnv, DateMarkerMeta, DateFormatter, createFormatter, VerboseFormattingArg, formatIsoTimeString, NamedTimeZoneImpl, parseMarker, EventSourceDef, EventSource, EventSourceHash, Interaction, InteractionSettings, interactionSettingsToStore, interactionSettingsStore, InteractionSettingsStore, PointerDragEvent, Hit, dateSelectionJoinTransformer, eventDragMutationMassager, EventDropTransformers, EventResizeJoinTransforms, ElementDragging, formatDate, formatRange, globalDefaults, config, RecurringType, ParsedRecurring, DragMetaInput, DragMeta, parseDragMeta, createPlugin, PluginDef, PluginDefInput, ViewPropsTransformer, ViewContainerModifier, reducerFunc, Action, CalendarState, CalendarComponentProps, DayHeader, computeFallbackHeaderFormat, renderDateCell, DaySeries, EventInteractionState, EventRenderRange, sliceEventStore, hasBgRendering, getElSeg, DayTable, DayTableSeg, DayTableCell, Slicer, SlicedProps, EventMutation, applyMutationToEventStore, Constraint, ConstraintInput, AllowFunc, isPropsValid, isInteractionValid, EventApi, requestJson, EventHandlerName, EventHandlerArgs, EventHandlerArg, EventRenderingChoice, NON_DATE_PROPS, DATE_PROPS, eventDefParserFunc, GenericHash, ClassNameInput, ElementContent, UNSCOPED_EVENT_UI_PROPS, default, ThemeClass, EqualityFuncHash, DateComponentHash, CalendarInteraction, CalendarInteractionClass, OptionChangeHandler, OptionChangeHandlerMap, ViewDefHash, ViewSpecHash, DAY_IDS, DurationInput, DateInput, FormatterInput, NamedTimeZoneImplClass, EventSourceError, EventInputTransformer, EventSourceSuccessResponseHandler, EventSourceErrorResponseHandler, EventSourceInput, EventSourceFetcher, InteractionClass, eventIsDraggableTransformer, ElementDraggingClass, rtlDefaults, ViewPropsTransformerClass, PluginSystem, CalendarComponent, eventDefMutationApplier, OverlapFunc, isPropsValidTester, LocaleCodeArg, LocaleSingularArg, RawLocaleMap, OptionsManager, EventSourceApi, ViewClass, ViewConfigInput, ViewConfigInputHash, ViewConfigHash, CmdFormatterFunc, CmdFormatter, FuncFormatterFunc, FuncFormatter, EventSourceFunc, _default, ExternalDefTransform, Toolbar
    
    5 declare module '@fullcalendar/core' {
      ~~~~~~~
    
      node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:5:1
        5 declare module '@fullcalendar/core' {
          ~~~~~~~
        Conflicts are in this file.
    node_modules/@fullcalendar/core/main.d.ts:115:9 - error TS2374: Duplicate string index signature.
    
    115         [viewId: string]: string | undefined;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:453:9 - error TS2374: Duplicate string index signature.
    
    453         [extendedProp: string]: any;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:970:9 - error TS2687: All declarations of 'view' must have identical modifiers.
    
    970         view: View;
                ~~~~
    node_modules/@fullcalendar/core/main.d.ts:1009:9 - error TS2717: Subsequent property declarations must have the same type.  Property 'component' must be of type 'DateComponent<any>', but here has type 'DateComponent<any>'.
    
    1009         component?: DateComponent<any>;
                 ~~~~~~~~~
    
      node_modules/@fullcalendar/angular/node_modules/@fullcalendar/core/main.d.ts:1012:9
        1012         component?: DateComponent<any>;
                     ~~~~~~~~~
        'component' was also declared here.
    node_modules/@fullcalendar/core/main.d.ts:1014:9 - error TS2374: Duplicate string index signature.
    
    1014         [otherProp: string]: any;
        [0[39mm         ~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:1494:9 - error TS2374: Duplicate string index signature.
    
    1494         [otherProp: string]: any;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:1503:9 - error TS2374: Duplicate string index signature.
    
    1503         [otherProp: string]: any;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:1799:9 - error TS2374: Duplicate string index signature.
    
    1799         [otherProp: string]: any;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:2580:9 - error TS2374: Duplicate string index signature.
    
    2580         [otherProp: string]: any;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/@fullcalendar/core/main.d.ts:2631:9 - error TS2374: Duplicate string index signature.
    
    2631         [optionName: string]: any;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~
    


```

Thanks!

**Comment by irustm** [comment]
@RicardoFazzi 
please run these two commands. 
```
npm ls @fullcalendar/core

and

npm ls @fullcalendar/angular
```

**Comment by domenichelfenstein** [comment]
> the `@fullcalendar/angular@4.4.5-beta` package has been published with all the latest changes from the angular9 branch

I can confirm that the published beta-build runs flawlessly in my project. Great job!

(I have no clue why I couldn't get my own build of this branch to work with my project, but I'm happy the published build works)

**Comment by steinv** [comment]
Same here! `@fullcalendar/angular@4.4.5-beta` works perfectly for my angular9 project.

**Comment by Huluti** [comment]
Do you have any idea when fullcalendar will support Angular 9 in its stable version?

**Comment by arshaw** [comment]
FYI, I've released a second v**5** beta. blog post: http://fullcalendar.test/blog/2020/04/new-ui-features-beta

[Version 5.0.0-beta.2](https://www.npmjs.com/package/@fullcalendar/angular/v/5.0.0-beta.2) of the angular connector is available.

It uses Angular 9.

The next stable version of FullCalendar to support Angular 9 will be v5. We won't add support to v4. So, plz just wait until v5 is officially released.

**Comment by arshaw** [comment]
this has been [officially released in v5](https://fullcalendar.io/blog/2020/06/v5-official)

please make sure you read the updated angular instructions here:
https://fullcalendar.io/docs/angular

please pay attention to `registerPlugins`

**Pull request merged**